core/src/main/scala-3/medeia/generic/GenericDecoderInstances.scala

```diff
@@ -1,13 +1,107 @@
 package medeia.generic
 
+import cats.data.*
+import cats.syntax.either.*
+import shapeless3.deriving.K0.*
 import shapeless3.deriving.*
+import org.mongodb.scala.bson.*
+import scala.compiletime.*
+import medeia.decoder.BsonDecoderError.*
+import medeia.decoder.BsonDecoderError
+import medeia.decoder.BsonDecoder
+import medeia.decoder.StackFrame.*
+import medeia.syntax.*
 
-trait GenericDecoderInstances {
-  @SuppressWarnings(Array("org.wartremover.warts.ImplicitConversion"))
-  implicit inline def genericDecoder[A](using
-      gen: K0.Generic[A],
-      gdOptions: GenericDerivationOptions[A] = GenericDerivationOptions[A](),
-      stOptions: SealedTraitDerivationOptions[A] = SealedTraitDerivationOptions[A]()
-  ): GenericDecoder[A] =
-    gen.derive(GenericProductDecoder.decoder, GenericCoproductDecoder.decoder)
+private[medeia] trait GenericDecoderInstances {
+  inline def apply[A]: GenericDecoder[A] =
+    summonInline[GenericDecoder[A]]
+
+  given coproduct[A](using
```

Comment:
From here downwards the changes in this file ares mostly code moved from the deleted files. I added `ProductDecoder` and it's usage but the rest mostly stayed the same.